jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Attributes.java

```diff
@@ -67,18 +67,25 @@
     // TODO something better than this
     default Map<String, Object> asAttributeMap()
     {
+        final Set<String> attributeNameSet = getAttributeNameSet();
```

Comment:
Since you have moved the `names` iterator to be a field, also move the names set to be a field of the anonymous abstract map.